File: app/src/main/java/com/onepointsixtwo/daggerviewmodelexample/MainActivity.java
package com.onepointsixtwo.daggerviewmodelexample;

import android.os.Bundle;
import android.widget.TextView;

import androidx.fragment.app.FragmentActivity;

import com.onepointsixtwo.dagger_viewmodel.InjectViewModel;

public class MainActivity extends FragmentActivity {

    @InjectViewModel
    public TestViewModel testViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume() {
        super.onResume();

        TextView textView = findViewById(R.id.text_view);
        textView.setText(testViewModel.getValueFromDependency());
    }
}

File: app/src/main/java/com/onepointsixtwo/daggerviewmodelexample/MainActivity.kt
package com.onepointsixtwo.daggerviewmodelexample

import android.os.Bundle
import android.widget.TextView

import androidx.fragment.app.FragmentActivity

import com.onepointsixtwo.dagger_viewmodel.InjectViewModel

class MainActivity : FragmentActivity() {

    @field:InjectViewModel
    lateinit var testViewModel: TestViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()

        val textView = findViewById<TextView>(R.id.text_view)
        textView.text = testViewModel.getValueFromDependency()
    }
}
